### PR TITLE
Adoc export

### PR DESCRIPTION
**Pull request by dragonchaser**
This PR adds support for generating adoc files for the environment variables for consumption in the docs.
Version information is still missing (but prepared) and needs to be annotated with the variables and output needs to be added to the template.

/cc @mmattel 

Refs #3583

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-2</strong>](https://drone.owncloud.com/owncloud/ocis/11284/25/1) failed. Further test are cancelled...

File: docs/templates/ADOC.tmpl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[caption=]
+.Environment variables for the frontend extension
```
Comment (mmattel) [inline docs/templates/ADOC.tmpl:2]: `.Environment variables for the frontend extension`

The name `frontend` needs to be replaced by the name of the extension the file is created for. Another example would be app-provider ect...

**Comment by butonic** [comment]
@dragonchaser how can I test this?

File: docs/templates/ADOC.tmpl (hunk 1 above)
Comment (dragonchaser) [reply docs/templates/ADOC.tmpl]: Adapted in https://github.com/owncloud/ocis/pull/3659/commits/4e0e19dfc7d4a6db4d036c76cf36b70e29736589

**Comment by dragonchaser** [comment]
> @dragonchaser how can I test this?

goto `docs/helpers`
run: `go run configextractor.go`

**Comment by dragonchaser** [comment]
> I would add the version field as it was in the original - even it will not get populated automatically atm (no generation code, no field in the go code), but it will force us to make this happen as the field is visible... You may also, for the time being, add a default value like `+1.20.0` (or whatever the beta version will be) as done in the original adoc file.

this is already prepared, just not output in the template to avoid confusion

**Review by mmattel — APPROVED** [review]
LGTM

**Pull request merged**